cyral/resource_cyral_access_rules.go

```diff
@@ -0,0 +1,329 @@
+	}
+
+	arr.AccessRules = accessRules
+	return nil
+}
+
+var ReadRepositoryAccessRulesConfig = ResourceOperationConfig{
+	Name:       "RepositoryAccessRulesRead",
+	HttpMethod: http.MethodGet,
+	CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+		return fmt.Sprintf("https://%s/v1/repos/%s/userAccounts/%s/accessRules",
+			c.ControlPlane,
+			d.Get("repository_id").(string),
+			d.Get("user_account_id").(string),
+		)
+	},
+	NewResponseData: func(_ *schema.ResourceData) ResponseData {
+		return &AccessRulesResponse{}
+	},
+}
+
+func resourceRepositoryAccessRules() *schema.Resource {
+	return &schema.Resource{
+		Description: "Manage access rules",
+		CreateContext: CreateResource(
+			ResourceOperationConfig{
+				Name:       "RepositoryAccessRulesCreate",
+				HttpMethod: http.MethodPost,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+					repoID := d.Get("repository_id").(string)
+					userAccountID := d.Get("user_account_id").(string)
+					d.SetId(
+						marshalComposedID(
+							[]string{
+								repoID,
+								userAccountID,
+							},
+							"/",
+						),
+					)
+					return fmt.Sprintf("https://%s/v1/repos/%s/userAccounts/%s/accessRules",
+						c.ControlPlane,
+						repoID,
+						userAccountID,
+					)
+				},
+				NewResourceData: func() ResourceData {
+					return &AccessRulesResource{}
+				},
+				NewResponseData: func(_ *schema.ResourceData) ResponseData {
+					return &AccessRulesResponse{}
+				},
+			},
+			ReadRepositoryAccessRulesConfig,
+		),
+		ReadContext: ReadResource(ReadRepositoryAccessRulesConfig),
+		UpdateContext: UpdateResource(
+			ResourceOperationConfig{
+				Name:       "RepositoryAccessRulesUpdate",
+				HttpMethod: http.MethodPut,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+					return fmt.Sprintf("https://%s/v1/repos/%s/userAccounts/%s/accessRules",
+						c.ControlPlane,
+						d.Get("repository_id").(string),
+						d.Get("user_account_id").(string),
+					)
+				},
+				NewResourceData: func() ResourceData {
+					return &AccessRulesResource{}
+				},
+				NewResponseData: func(_ *schema.ResourceData) ResponseData {
+					return &AccessRulesResponse{}
+				},
+			},
+			ReadRepositoryAccessRulesConfig,
+		),
+		DeleteContext: DeleteResource(
+			ResourceOperationConfig{
+				Name:       "RepositoryAccessRulesDelete",
+				HttpMethod: http.MethodDelete,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+
+					idPieces, err := unmarshalComposedID(d.Id(), "/", 2)
+					if err != nil {
+						panic(fmt.Sprintf("Failed to unmarshal access rules ID: %v", err))
+					}
+					repoID := idPieces[0]
+					userAccountID := idPieces[1]
+
+					return fmt.Sprintf("https://%s/v1/repos/%s/userAccounts/%s/accessRules",
+						c.ControlPlane,
+						repoID,
+						userAccountID,
+					)
+				},
+			},
+		),
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description: "ID of this resource in Cyral environment.",
```

Comment:
I think would be good to describe how the ID is composed (E.g.: {repository_id}/{user_account_id}), or we could just include a note on the documentation similar to what's done for the identity map resource (https://github.com/cyralinc/terraform-provider-cyral/blame/main/docs/resources/repository_identity_map.md#L5)). This would help users to understand how they should be importing the resource.

Reply:
Yeah, I think having this in the docs precisely how we do it for that stuff you linked will be the cleanest way to do this.